### PR TITLE
PR: Fix find/replace widget for multiline regex

### LINKED ISSUE #2675: Multiline search and replace doesn't replace
## Description

**What steps will reproduce the problem?**
1. Activate search & replace widget
2. With regular expression mode engaged, search for a multiline pattern (e.g. \n\n\n)
3. Enter some text to replace the expression, hit replace/find button.

**What is the expected output? What do you see instead?**

You would expect the multiline search pattern to be replaced with the replacement text. Instead, no replacement is made.

**Please provide any additional information below**
## Version and main components
- Spyder Version: 2.3.6 
- Python Version: 2.7.9
- Qt Versions:  4.8.6, PyQt4 (API v2) 4.10.4 on Windows
## Optional dependencies

```
pyflakes >=0.5.0:  0.9.2 (OK)
pep8 >=0.6      :  1.6.2 (OK)
pygments >=1.6  :  2.0.2 (OK)
IPython >=1.0   :  4.0.0 (OK)
zmq >=2.1.11    :  14.7.0 (OK)
qtconsole >=4.0 :  4.0.1 (OK)
pandas >=0.13.1 :  0.16.2 (OK)
sphinx >=0.6.6  :  1.3.1 (OK)
rope >=0.9.2    :  0.9.4 (OK)
jedi >=0.8.1    :  0.9.0 (OK)
matplotlib >=1.0:  1.4.3 (OK)
sympy >=0.7.3   :  0.7.6 (OK)
pylint >=0.25   :  None (NOK)

```

### PR DESCRIPTION
**Pull request by goanpeca**
### Issue(s) Resolved

Fixes #2675

### Affirmation

By submitting this Pull Request or typing my (user)name below,
I affirm the [Developer Certificate of Origin](https://developercertificate.org)
with respect to all commits and content included in this PR,
and understand I am releasing the same under Spyder's MIT (Expat) license.

I certify the above statement is true and correct: @goanpeca

### CODE REVIEW

File: spyder/widgets/findreplace.py [hunk 1]
```diff
@@ -527,23 +528,34 @@ def replace_find(self, focus_replace_text=False, replace_all=False):
...
                 cursor.removeSelectedText()
                 cursor.insertText(replace_text)
             else:
                 seltxt = to_text_string(cursor.selectedText())
+
+                # Note: If the selection obtained from an editor spans a line
+                # break, the text will contain a Unicode U+2029 paragraph
+                # separator character instead of a newline \n character.
+                # See: spyder-ide/spyder#2675
+                seltxt = seltxt.replace(u'\u2029', u'\n')
```
Comment (ccordoba12) [inline spyder/widgets/findreplace.py]: Should we use here the EOL characters used in the file, instead of `\n`?

Comment (goanpeca) [reply spyder/widgets/findreplace.py]: Hmmmm good point. Are the EOF chars handled in real time or  at the moment of saving the file?

Comment (ccordoba12) [reply spyder/widgets/findreplace.py]: I think you could use the function `get_eol_chars` (present on `utils/sourcecode.py`) on `seltxt` to get them.

**Comment by ccordoba12** [comment]
@goanpeca, please finish this PR to include it in 4.0.1

**Comment by goanpeca** [comment]
Will do later today 👍

**Comment by goanpeca** [comment]
Fixed @ccordoba12

**Comment by goanpeca** [comment]
Added test for this @ccordoba12

**Pull request merged**